packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -247,10 +264,15 @@ function createHook(): Hook {
 
     next: null,
   };
+  if (__DEV__) {
+    invariant(currentHookType !== null, 'Forgot to set currentHookType');
```

Comment:
I might just remove this

Reply:
I think its safe to remove, a comment is probably better? Writing to a top-level variable is a pretty common pattern in the codebase, and new hook types are rare enough that a comment is probably a sufficient safeguard